### PR TITLE
dnsbench rewrite

### PR DESCRIPTION
**Pull request by richardmarshall**
Major rewrite of dnsbench.

**Goals**

- Improve split between CLI and benchmarking logic to enable usage as a package in other tools.
- Move to a more intuitive QPS target model.
- Ease of testing.
- More flexible interfaces to enable development of new features such as TCP and TLS support.
- Switch from dep to go modules.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,30 +15,25 @@ go get -u github.com/askmediagroup/dnsbench/cmd/dnsbench
...
-  -n, --count int           Total number requests. (default 1000)
+  -c, --count int           Number of queries to attempt. [0 = run until interrupted] (default 100)
   -h, --help                help for run
-  -i, --interval duration   Number of concurrent requests. (default 5s)
-  -f, --names string        File containing newline delimited records to lookup. (- for stdin) (default "-")
-  -q, --qps int             QPS target for each concurrent worker.
-
-Use "dnsbench run [command] --help" for more information about a command.
+  -i, --interval duration    (default 5s)
+  -m, --max-workers uint    . (default 10)
```
Comment (makhadi) [inline README.md]: remove `. `(dot & space)

File: pkg/namer_test.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+package dnsbench
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestNamer(t *testing.T) {
+	names := []string{"github.com", "gitlab.com", "bitbucket.org"}
+	expectedNames := []string{"github.com.", "gitlab.com.", "bitbucket.org.", "github.com."}
```
Comment (makhadi) [inline pkg/namer_test.go:13]: this can be 3 names and the same at #21(remove duplicate)?

Comment (richardmarshall) [reply pkg/namer_test.go:13]: This is actually intentional to test that the namer will cycle back to the beginning of the names list after running through to the end. 

If you see below we're calling `namer()` 4 times so the expected output needs to include 4 items.

Comment (makhadi) [reply pkg/namer_test.go:13]: ok 👍

**Comment by makhadi** [comment]
looks good overall, if you can make formatting change and remove draft

can merge if @yoitsbenco don't have any comments

File: README.md (hunk 1 above)
Comment (richardmarshall) [reply README.md]: I need to regenerate this help example, fixed up those flags to have descriptions in the code but didn't update the readme.

Comment (richardmarshall) [reply README.md]: Fixed.

**Comment by richardmarshall** [comment]
Cleaned up the branch history. @yoitsbenco if you have any comments would appreciate them before this is merged.

File: README.md [hunk 3]
```diff
@@ -15,30 +15,27 @@ go get -u github.com/askmediagroup/dnsbench/cmd/dnsbench
...
-  local       Benchmark the local system resolver configuration.
-  remote      Benchmark a remote nameserver.
+  dnsbench run [command] [flags]
 
 Flags:
-  -c, --concurrency int     Number of concurrent requests. (default 10)
-  -n, --count int           Total number requests. (default 1000)
+  -c, --count int           Number of queries to attempt. [0 = run until interrupted] (default 100)
   -h, --help                help for run
-  -i, --interval duration   Number of concurrent requests. (default 5s)
```
Comment (benco5) [inline README.md:31]: ooh! was this a previous typo/mistake? @richardmarshall

Comment (richardmarshall) [reply README.md:31]: Yep, looks like an old copy&paste error.

**Comment by richardmarshall** [comment]
Now that I'm back from hiatus I'm going to go ahead and just merge this PR myself.

**Pull request merged**